testing/src/main/java/io/opencensus/testing/propagation/HttpUtils.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2018, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opencensus.testing.propagation;
+
+/**
+ * Utilities for HTTP testing.
+ *
+ * @since 0.13
+ */
+public final class HttpUtils {
+
+  private HttpUtils() {}
+
+  /**
+   * Returns {@code false} if a header name contains invalid characters, otherwise {@code true}.
+   *
+   * <p>Specification for HTTP header name: See RFC-7230, which explicitly defines what characters
+   * are accepted. Note that RFC-2616 has different BNF rules on header name, but the derived
+   * character set is the same.
+   *
+   * <pre>
+   * field-name = token
+   * token      = 1*tchar
+   * tchar      = "!" / "#" / "$" / "%" / "&amp;" / "'" / "*" / "+" / "-" / "." / "^" / "_"
+   *            / "`" / "|" / "~" / DIGIT / ALPHA
+   * </pre>
+   *
+   * <p>This method is also inspired by
+   * https://github.com/netty/netty/blob/fc3b145cbbcad4a257f92f0640eb4ded19b44afe/codec-http/src/main/java/io/netty/handler/codec/http/DefaultHttpHeaders.java#L381:25
```

Comment:
Is there any way that we can reuse the method from Netty (or another library), since this is only used in tests, where we have fewer restrictions on dependencies?

Reply:
The Netty one does not fully follow the spec, though it should be enough to serve the purpose of preventing similar issues. Though the method itself is private, we can construct a `DefaultHttpHeaders` if we want to trigger the validation logic.

I am fine with using Netty if dependency is not a concern.

Reply:
I can't think of any problems with adding Netty as a test-only dependency, except possibly slowing down the build.

Using Netty would also work around the issue of needing to share code between two test suites.  We'll probably still need to find a way to share code at some point, though.